core/src/test/java/org/dcsa/conformance/core/check/JsonAttributeTest.java

```diff
@@ -0,0 +1,732 @@
+    objectNode.put("test", "test");
+    assertTrue(JsonAttribute.isEqualTo("test", "test").test(objectNode));
+    assertFalse(JsonAttribute.isEqualTo("test", "testFalse").test(objectNode));
+  }
+
+  @Test
+  void testIsOneOf() {
+    ObjectNode objectNode = JsonNodeFactory.instance.objectNode();
+    objectNode.put("test", "test");
+
+    assertTrue(JsonAttribute.isOneOf("test", Set.of("test", "test1")).test(objectNode));
+    assertFalse(JsonAttribute.isOneOf("test", Set.of("test2", "test1")).test(objectNode));
+  }
+
+  @Test
+  void testMustBePresent() {
+    ObjectNode objectNode = JsonNodeFactory.instance.objectNode();
+    objectNode.put("test", "value");
+
+    assertTrue(
+        JsonAttribute.mustBePresent(JsonPointer.compile("/test"))
+            .validate(objectNode, "")
+            .isEmpty());
+    assertFalse(
+        JsonAttribute.mustBePresent(JsonPointer.compile("/nonexistent"))
+            .validate(objectNode, "")
+            .isEmpty());
+  }
+
+  @Test
+  void testIfThen() {
+    ObjectNode objectNode = JsonNodeFactory.instance.objectNode();
+    objectNode.put("test", "test");
+    objectNode.put("test2", "test2");
+    assertTrue(
+        JsonAttribute.ifThen(
+                "Test ifThen with correct value",
+                JsonAttribute.isNotNull(JsonPointer.compile("/test")),
+                JsonAttribute.mustBePresent(JsonPointer.compile("/test2")))
+            .validate(objectNode, "")
+            .isEmpty());
+
+    assertFalse(
+        JsonAttribute.ifThen(
+                "Test ifThen with incorrect value",
+                JsonAttribute.isNotNull(JsonPointer.compile("/test")),
+                JsonAttribute.mustBePresent(JsonPointer.compile("/test3")))
+            .validate(objectNode, "")
+            .isEmpty());
+
+    // Test when initial condition is false
+    assertTrue(
+        JsonAttribute.ifThen(
+                "Test ifThen with initial condition false",
+                JsonAttribute.isNotNull(JsonPointer.compile("/test3")),
+                JsonAttribute.mustBePresent(JsonPointer.compile("/test2")))
+            .validate(objectNode, "")
+            .isEmpty());
+  }
+
+  static Stream<Arguments> testIsNotNullArgs() {
+    return Stream.of(
+        Arguments.of("test isNotNull", JsonNodeFactory.instance.objectNode(), "/test", false),
+        Arguments.of(
+            "test isNotNull with existing value",
+            JsonNodeFactory.instance.objectNode().put("test", "test"),
+            "/test",
+            true));
+  }
+
+  @ParameterizedTest(name = "[{index}] {0}")
+  @MethodSource("testIsNotNullArgs")
+  void testIsNotNull(String testName, ObjectNode input, String path, boolean expected) {
+    assertEquals(expected, JsonAttribute.isNotNull(JsonPointer.compile(path)).test(input));
+  }
+
+  @Test
+  void testMustBeNotNull() {
+    ObjectNode objectNode = JsonNodeFactory.instance.objectNode();
+    objectNode.put("test", "test");
```

Comment:
These 2 sentences are repeated 18 times. Please put them in a `before()` method, so JUnit runs them automatically before every invocation of a test. And make `objectNode` a variable of this class.

Reply:
ObjectNode objectNode = JsonNodeFactory.instance.objectNode(); line is moved to BeforeEach method.